### PR TITLE
Revisit the [UIScrollView _adjustContentOffsetIfNecessary] issue.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 1]
```diff
@@ -269,17 +269,6 @@ - (void)viewDidLoad
...
-    // Per https://twitter.com/steipete/status/219848151765483520?lang=en
-    // "Is there some trick to bring UINavigationController to not mess with a UIScrollView  (_adjustContentOffsetIfNecessary)?"
-    // "Add a dummy view as first subview. It won’t look further."
-    //
-    // This hack prevents UINavigationController from calling _adjustContentOffsetIfNecessary
-    // on the collection view after viewWillAppear: but before viewDidAppear:, which
-    // causes the scroll state to reset to zero.    
-    UIView *dummyView = [UIView new];
-    dummyView.hidden = YES;
-    [self.view insertSubview:dummyView atIndex:0];
```
Comment (charlesmchen) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m:282]: This hack only addressed one of the triggers of `[UIScrollView _adjustContentOffsetIfNecessary]`.  There are many others.  We need a more comprehensive fix.

File: JSQMessagesViewController/Views/JSQMessagesCollectionView.m [hunk 2]
```diff
@@ -37,6 +37,29 @@ - (void)jsq_configureCollectionView;
...
+        // response to many different events; trying to prevent them all is
+        // whack-a-mole.
+        //
+        // It's not safe to override [UIScrollView _adjustContentOffsetIfNecessary],
+        // since its a private API.
+        //
+        // We can avoid the issue by simply ignoring attempt to reset the content
+        // offset to zero before the collection view has determined its content size.
+        return;
+    }
```
Comment (charlesmchen) [inline JSQMessagesViewController/Views/JSQMessagesCollectionView.m:58]: The new fix.

Comment (michaelkirk) [reply JSQMessagesViewController/Views/JSQMessagesCollectionView.m:58]: This **seems** reasonable enough. 

The goal here is staying scrolled to bottom, right?

**Review by michaelkirk — APPROVED** [review]
LGTM

**Comment by charlesmchen** [comment]
> The goal here is staying scrolled to bottom, right?

The goal here (to be very specific) is let message view control _initial_ scroll state and not have that overwritten wily-nilly by iOS.  That might be bottom, or it might be "show the unread indicator and the stuff below it", etc.

**Pull request merged**